docs/source/java/flight.rst

```diff
@@ -0,0 +1,216 @@
+
+To connect to a Flight service, call `FlightClient`_ with a location.
+
+.. code-block:: Java
+
+    Location location = Location.forGrpcInsecure("0.0.0.0", 58104);
+
+    try(BufferAllocator allocator = new RootAllocator();
+        FlightClient client = FlightClient.builder(allocator, location).build()){
+        // ... Consume operations exposed by Flight server
+    } catch (Exception e) {
+        e.printStackTrace();
+    }
+
+Cancellation and Timeouts
+=========================
+
+When making a call, clients can optionally provide `CallOptions`. This allows
+clients to set a timeout on calls. Also, some objects returned by client RPC calls
+expose a cancel method which allows terminating a call early.
+
+.. code-block:: Java
+
+    Location location = Location.forGrpcInsecure("0.0.0.0", 58609);
+
+    try(BufferAllocator allocator = new RootAllocator();
+        FlightClient tutorialFlightClient = FlightClient.builder(allocator, location).build()){
+
+        Iterator<Result> resultIterator = tutorialFlightClient.doAction(
+                new Action("test-timeout"),
+                CallOptions.timeout(2, TimeUnit.SECONDS)
+        );
+    } catch (Exception e) {
+        e.printStackTrace();
+    }
+
+On the server side, timeouts are transparent. For cancellation, the server needs to manually poll
+`setOnCancelHandler` or `isCancelled` to check if the client has cancelled the call, and if so,
+break out of any processing the server is currently doing.
+
+.. code-block:: Java
+
+    // Client
+    Location location = Location.forGrpcInsecure("0.0.0.0", 58609);
+    try(BufferAllocator allocator = new RootAllocator();
+        FlightClient tutorialFlightClient = FlightClient.builder(allocator, location).build()){
+        try(FlightStream flightStream = flightClient.getStream(new Ticket(new byte[]{}))) {
+            // ...
+            flightStream.cancel("tutorial-cancel", new Exception("Testing cancellation opion!"));
+        }
+    } catch (Exception e) {
+        e.printStackTrace();
+    }
+    // Server
+    @Override
+    public void getStream(CallContext context, Ticket ticket, ServerStreamListener listener) {
+        // ...
+        listener.setOnCancelHandler(()->{
+                    // Implement logic to handle cancellation option
+                });
+    }
+
+Enabling TLS
+============
+
+TLS can be enabled when setting up a server by providing a
+certificate and key pair to `FlightServer.builder.useTls`.
+
+On the client side, use `FlightClient.builder.trustedCertificates`.
+
+Enabling Authentication
+=======================
+
+.. warning:: Authentication is insecure without enabling TLS.
+
+Handshake-based authentication can be enabled by implementing
+`ServerAuthHandler`. Authentication consists of two parts: on
+initial client connection, the server and client authentication
+implementations can perform any negotiation needed; then, on each RPC
+thereafter, the client provides a token. The server authentication
```

Comment:
Can you look at the wording in #12815 again? This description is inaccurate

Reply:
Changed

Reply:
This is still different - this doesn't describe where the client token comes from.

Reply:
Added where the token comes from